#VRML_SIM R2021b utf8
WorldInfo {
  gravity 0
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9238082126993563 0.37733661600553237 0.0647724043919481 0.3674381043345132
  position 0.18631605522979605 0.18644789449169857 0.9396144681309498
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Robot {
  children [
    DEF CLASSIC_HINGE HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
      }
      device [
        PositionSensor {
          name "cJointPos"
        }
        RotationalMotor {
          name "cJointMot"
        }
      ]
      endPoint Solid {
        translation -6.75616e-11 -0.15 -0.1
        rotation 0 0 1 0
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0.8 0 0
            }
            geometry DEF PENDULUM Box {
              size 0.02 0.3 0.1
            }
          }
        ]
        name "solid(1)"
        boundingObject USE PENDULUM
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF AXIS1 HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
      }
      device [
        PositionSensor {
          name "hJointPos"
        }
        RotationalMotor {
          name "hJointMot"
        }
      ]
      endPoint Solid {
        translation -1.8014757060106203e-06 -1.4914922116814823e-09 0
        rotation 0 0 1 4.3683953004699476e-05
        children [
          Transform {
            rotation 1 0 0 -1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                }
                geometry Cylinder {
                  height 0.5
                  radius 0.001
                }
              }
            ]
          }
          DEF AXIS2 HingeJoint {
            jointParameters HingeJointParameters {
              axis 0 0 1
              minStop -0.1
              maxStop 0.1
              stopERP 0
            }
            device [
              PositionSensor {
                name "hJointPos2"
              }
            ]
            endPoint Solid {
              translation 1.2697599915956204e-11 -0.14999999999999997 0
              rotation 0 0 1 0
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.203922 0.396078 0.643137
                  }
                  geometry USE PENDULUM
                }
              ]
              boundingObject USE PENDULUM
              physics Physics {
                density -1
                mass 1
              }
            }
          }
        ]
        boundingObject Sphere {
          radius 0.01
        }
        physics Physics {
          density -1
          mass 0.001
        }
      }
    }
    HingeJointWithBacklash {
      axis 0 0 1
      backlash 0.2
      device [
        RotationalMotor {
          name "pJointMot"
        }
      ]
      endPoint Solid {
        translation 0 -0.15 0.1
        rotation 0 0 1 0
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0.45098 0.823529 0.0862745
            }
            geometry USE PENDULUM
          }
        ]
        boundingObject USE PENDULUM
        physics Physics {
          density -1
          mass 1
        }
      }
      startPoint [
        Transform {
          translation 0 0 0.1
          rotation 1 0 0 -1.57
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.305882 0.603922 0.0235294
              }
              geometry Box {
                size 0.05 0.06 0.05
              }
            }
          ]
        }
      ]
    }
  ]
  name "robot(1)"
  controller "hingewithbacklash"
}
